R/geotargets-option.R

```diff
@@ -0,0 +1,65 @@
+#' Get or Set geotargets Options
+#'
+#' Get or set behavior for geospatial data target stores using geotargets-specific global options.
+#'
+#' @param option_name Character. Option name. See Details.
+#'
+#' @details
+#'
+#' ## Available Options
+#'
+#'  - `"geotargets.gdal.raster.creation_options"` - set the GDAL creation options used when writing raster files to target store (default: `"ENCODING=UTF-8"`)
+#'
+#'  - `"geotargets.gdal.raster.driver_name"` - set the file type used for raster data in target store (default: `"GTiff"`)
+#'
+#'  Each option can be overridden with a system environment variable. Options include:
+#'
+#'   - `GEOTARGETS_GDAL_RASTER_CREATION_OPTIONS`
+#'   - `GEOTARGETS_GDAL_RASTER_DRIVER_NAME`
+#'
+#' @rdname geotargets-options
+#' @export
+geotargets_option_get <- function(option_name) {
+    if (!startsWith(option_name, "geotargets.")) {
+        option_name <- paste0("geotargets.", option_name)
+    }
+
```

Comment:
```suggestion
targets:::tar_assert_flag(option_name, choices = names(geotargets.env))
```
Or some other method of erroring when an invalid option name is provided

Reply:
Good idea!

```r
rlang::arg_match0(option_name, names(geotargets.env)
```

should work, and means we won't need to do a (CRAN illegal) `targets:::`

Reply:
broke this out into a separate issue: https://github.com/njtierney/geotargets/issues/27